pkg/verification/verification_service.go

```diff
@@ -0,0 +1,87 @@
+package verification
+
+import (
+	"crypto/rand"
+	"fmt"
+	"net/http"
+
+	"github.com/codeready-toolchain/api/pkg/apis/toolchain/v1alpha1"
+	"github.com/codeready-toolchain/registration-service/pkg/log"
+	"github.com/gin-gonic/gin"
+	"github.com/kevinburke/twilio-go"
+)
+
+const (
+	codeCharset = "0123456789"
+	codeLength  = 6
+)
+
+// ServiceConfiguration represents the config used for the verification service.
+type ServiceConfiguration interface {
+	GetTwilioAccountSID() string
+	GetTwilioAuthToken() string
+	GetTwilioFromNumber() string
+	GetVerificationMessageTemplate() string
+}
+
+// Service represents the verification service for controllers.
+type Service interface {
+	SendVerification(ctx *gin.Context, signup *v1alpha1.UserSignup) error
+}
+
+// ServiceImpl represents the implementation of the verification service.
+type ServiceImpl struct {
+	config     ServiceConfiguration
+	HttpClient *http.Client
+}
+
+type VerificationServiceOption func(svc *ServiceImpl)
+
+// NewVerificationService creates a service object for performing user verification
+func NewVerificationService(cfg ServiceConfiguration, opts ...VerificationServiceOption) Service {
```

Comment:
You are not initiating any default http client? Looks a bit error prone. Http client is not among the required params but it is actually required.

Reply:
It's actually not required.  The very first thing that the `NewClient()` function does is create a default http client if it isn't set:

```
func NewClient(accountSid string, authToken string, httpClient *http.Client) *Client {
	if httpClient == nil {
		httpClient = defaultHttpClient
	}
```

Reply:
All good then!